R/moduleInstall.R

```diff
@@ -361,15 +364,648 @@ setupRenv <- function(moduleLibrary, modulePkg) {
+    options(oldOptions)
+  }, add = TRUE, after = FALSE)
+
+  r <- getOption("repos")
+  r["CRAN"] <- repos
+  options(repos = r)
+
+  setupRenv(moduleLibrary, modulePkg)
+
+  return(pkgbuild::with_build_tools(
+    installModuleNew(modulePkg, jaspRoot, moduleLibrary, updatePackages = updatePackages, recordPackages = recordPackages,
+                     respectModuleLockfile = respectModuleLockfile),
+    required = FALSE
+  ))
+
+}
+
+map <- function(x, f, ...) {
+  res <- lapply(x, f, ...)
+  names(res) <- x
+  res
+}
+
+
+#' Install a Jasp module and update its dependencies.
+#'
+#' @param modulePath local path to jasp module, e.g., ~/github/jasp-desktop/Modules/jaspDescriptives (note that this does not need to be inside jasp-desktop)
+#' @param jaspRoot local path to clone of jasp-desktop, e.g., ~/github/jasp-desktop
+#' @param moduleLibrary where to install the module
+#' @param lockfilePath optional, path to a lockfile.
+#' @param updatePackages optional, whether to update the non JASP R package dependencies. Defaults to false, but see Details.
+#' @param recordPackages optional, whether to record all package dependencies as they are installed ("all"), or whether to change records of Jasp modules to the locally checked out version ("localJasp").
+#' @param recurseJaspDependencies optional, whether to use the local versions of indirect Jasp module dependencies. Defaults to TRUE.
+#' @param useLocalLockfile optional, whether to use the lockfile present in modulePath. Defaults to false.
+#' @param prompt optional, show renv promts. Ignored when not running in interactive mode (e.g., inside Jasp).
+#'
+#' @details \code{installModuleNew} does four things:
+#' 1. Install a Jasp module from scratch using locally checked out jasp modules. When `recurseJaspDependencies` is TRUE (the default), then it is ensured that any indirect dependencies on other jasp modules als use the locally checked out versions.
+#' 2. Update the Jasp module dependencies when locally checked out sources change. For example, when jaspBase is updated but jaspDescriptives not, this function ensures that jaspDescriptives will use the new version of jaspBase.
+#' 3. Optionally, update non Jasp R package dependencies.
+#' 4. TODO: Optionally, install a Jasp module using the lockfile specified in the source location. This ignores any options set for the previous three options.
+#'
+#' Possible values for \code{updatePackages} are "true", "false", "daily", "triweekly", "biweekly", "weekly", "fortnightly", "monthly",
+#' "monday", "tuesday", ... "sunday", or a combination of weekdays, "monday;wednesday;friday".
+#' Anything value that is not understood is interpreted as FALSE (with a warning).
+#' The date of the previous update is determined by looking at the last date where \code{lockfilePath} was modified.
+#'
+#' @rdname installModuleNew
+#'
+#' @return returns \code{NULL}.
+#'
+installModuleNew <- function(
```

Comment:
This function is so large I think it would be good to have some comments in between breaking up the flow describing what each section does. Like "Here we are going to scan the lockfile" or something.
It would help me, and probably future you, out a lot.

Splitting it up in some descriptively named functions would also be a good way to solve this but I can imagine that would be a bit painful so I wont ask for that.

Reply:
Will do!